zerver/lib/upload.py

```diff
@@ -844,6 +844,16 @@ def upload_avatar_image(user_file: File, acting_user_profile: UserProfile,
     upload_backend.upload_avatar_image(user_file, acting_user_profile,
                                        target_user_profile, content_type=content_type)
 
+def upload_avatar_image_from_url(avatar_url: str, user_profile: UserProfile) -> str:
+    import requests
```

Comment:
Can you move this import to the top of the file?  That's the usual way to import things; the only reason to do something different is to work around import cycles.